Migrate to setuptools, make python3 compatible, and reducing chattiness

kyle-walker opened the pull request:
Absolutely open to making changes to this PR, but a few of the changes included are:

 - Migrate to a [setuptools](https://setuptools.readthedocs.io/en/latest/) packaging strategy. The benefit being that we can then leverage the "python setup.py <verb>" workflow for packaging and installation.

- Made the utility python3 compliant, though this was done with a fairly heavy hand. Further work is remaining to help clean it up and make it more presentable.

- Corrected the references to the older github page with the new RedHatOfficial/rhsecapi URL.

- Reduced the chattiness of the utility outside of the query responses themselves. This translates mostly into setting a default logging level of WARNING so that we don't get the NOTICE messages unless we want them.

ryran committed:
fix #66

ryran committed:
readme update

kyle-walker committed:
all: Packaging for setuptools and making python3 compatible

Reorganizing the project. Multiple changes, but the intent of each
alteration is to make the project:

 - Installable via setuptools
 - Compatible with python2 and python3

kyle-walker committed:
setup: Change the formatting for RPM compatibility

Thre previous revision was exceptionally verbose and didn't play well with
the "python setup.py bdist_rpm" process. The new revision makes it more
terse.

kyle-walker committed:
rhsecapi: Remove the "less" pager use

There seems to be a problem within argparse for 3.7 which results in an
error around file handling. The process of resolving it would be too
arduous when compared to the alternative of avoiding calling "less"
directly in the manner previously used.

The recommended alternative is to simply pipe the help output to "less".

kyle-walker committed:
all: Change the "See <>" URL to reflect the new location

The previous location under "ryran" is no longer applicable. Moving
mentions of that path over to the RedHatOfficial group.

kyle-walker committed:
rhsecapi: Set the default log level to WARNING

The previous revision emitted a lot of informational output besides just
the response. This is helpful if the utility is being investigated, but
requiring a "-l warning" flag to omit those messages is not really
necessary.

Setting the default to WARNING to silence the NOTICE messages.

kyle-walker committed:
setup: Bump the version to 1.0.2

kyle-walker committed:
build: Add a rpkg build configuration and specfile

kyle-walker committed:
All: Remove IAVAs as a query subject

The backend security API no longer supports IAVAs as a query subject. This
was removed towards the end of 2018. This patch attempts to remove the
corresponding client-side IAVA query code while preserving the remaining
operations.

kyle-walker committed:
All: Remove the "python2" shebang callouts

Since we are now relying on the setuptools mechanism to install to either
python2 or python3, it no longer makes sense to include shebang callouts at
the top of the script.

kyle-walker commented:
It's been a few months, so - Bump!

kyle-walker commented:
(Quite) a few more months. - Bump!

digitronik commented:
@ryran is it possible to bump this?
Still love more pythonic love :)

kyle-walker committed:
.gitignore; Omit the .vscode directory

This avoids IDE configurations from being propagated into the project.

kyle-walker committed:
rhsecapi: Bump the version to 1.0.3

This updates the version both in the setup.py and also within the in-
CLI definition.

kyle-walker committed:
spec: Drop the python3-argparse Requires

It isn't necessary, and causes installation issues for systems where it
isn't available.

kyle-walker committed:
spec: Require the lib when installing the cli

The previous version didn't require the library, and so would fail with the
following error.

    Traceback (most recent call last):
       File "/usr/bin/rhsecapi", line 25, in <module>
        import rhsda
    ModuleNotFoundError: No module named 'rhsda'

kyle-walker committed:
rhsecapi: Fix the version

It wasn't updated with the last tag.

wyrmiyu commented:
Hi, any chance this could be merged at some point? The Python 3 support would be most desirable. Is there still something missing?

malvads commented:
Any updates on this?

kyle-walker committed:
git: Add .venv to .gitignore

Just making it simpler for those that use hidden cwd-venvs

kyle-walker committed:
packaging: Add a requirements.txt

A similar quality-of-life improvement allowing:

    pip install -r requirements.txt

kyle-walker committed:
packaging: Bump the version to 1.0.5

kyle-walker committed:
packaging: Remove rpkg entirely

The project seems to be unmaintained upstream, so I'm dropping it here in favor
of tito.

kyle-walker committed:
Initialized to use tito.

kyle-walker committed:
Automatic commit of package [rhsecapi] release [1.0.6-1].

Created by command:

/usr/bin/tito tag

kyle-walker committed:
Merge pull request #1 from kyle-walker/move-to-tito

Move to tito
